### PR TITLE
test(ticdc): fix region_merge test case

### PR DESCRIPTION
**Pull request by wk989898**
### What problem does this PR solve?

Issue Number: ref #11879 

### What is changed and how it works?
Ensuring "region merge" is expected

#### Tests 

 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: tests/integration_tests/region_merge/run.sh [hunk 1]
```diff
@@ -12,11 +12,16 @@ function split_and_random_merge() {
...
+	run_sql "ALTER TABLE region_merge.t1 ATTRIBUTES 'merge_option=deny';" ${UP_TIDB_HOST} ${UP_TIDB_PORT}
+	# region should be one
+	run_sql "SELECT count(distinct region_id) from information_schema.tikv_region_status where db_name = 'region_merge' and table_name = 't1';" &&
+		cat $OUT_DIR/sql_res.region_merge.txt
 	run_sql "SPLIT TABLE region_merge.t1 BETWEEN (-9223372036854775808) AND (9223372036854775807) REGIONS $scale;" ${UP_TIDB_HOST} ${UP_TIDB_PORT} || true
 	run_sql "SELECT count(distinct region_id) from information_schema.tikv_region_status where db_name = 'region_merge' and table_name = 't1';" &&
 		cat $OUT_DIR/sql_res.region_merge.txt
 	run_sql "insert into region_merge.t1 values (-9223372036854775808),(0),(1),(9223372036854775807);" ${UP_TIDB_HOST} ${UP_TIDB_PORT}
-	run_sql "delete from region_merge.t1;" ${UP_TIDB_HOST} ${UP_TIDB_PORT}
+	run_sql "truncate region_merge.t1;" ${UP_TIDB_HOST} ${UP_TIDB_PORT}
```
Comment (CharlesCheung96) [inline tests/integration_tests/region_merge/run.sh:23]: truncate changes the table id and cannot trigger the expected region merge

**Review by asddongmen — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**